Assets/SEE/XR/RadialSelection.cs

```diff
@@ -0,0 +1,409 @@
+    private void RadialMenu(InputAction.CallbackContext context)
+    {
+        radialMenuTrigger = true;
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+        if (radialMenuTrigger && !spawned)
+        {
+            SpawnRadialPart();
+            radialMenuTrigger = false;
+        }
+        if (spawned)
+        {
+            GetSelectedRadialPart();
+        }
+        if (radialMenuTrigger && spawned)
+        {
+            spawned = false;
+            radialMenuTrigger = false;
+            HideAndTriggerSelected();
+        }
+    }
+    /// <summary>
+    /// Is true, when the current action changed.
+    /// It is being used, to trigger the update for the GlobalActionHistory.
+    /// </summary>
+    public static bool IndicatorChange { set; get; }
+    /// <summary>
+    /// This method activates the selected action and deactivates the radial menu.
+    /// </summary>
+    public void HideAndTriggerSelected()
+    {
+        radialPartCanvas.gameObject.SetActive(false);
+        OnPartSelected.Invoke(currentSelectedRadialPart);
+        IndicatorChange = true;
+    }
+    /// <summary>
+    /// This method calculates, at which radial-part the user is aiming at.
+    /// </summary>
+    public void GetSelectedRadialPart()
+    {
+        Vector3 centerToHand = handTransform.position - radialPartCanvas.position;
+        Vector3 centerToHandProjected = Vector3.ProjectOnPlane(centerToHand, radialPartCanvas.forward);
+
+        float angle = Vector3.SignedAngle(radialPartCanvas.up, centerToHandProjected, -radialPartCanvas.forward);
+
+        if (angle < 0)
+        {
+            angle += 360;
+        }
+
+        currentSelectedRadialPart = (int)angle * numberOfRadialPart / 360;
+
+        for (int i = 0; i < spawnedParts.Count; i++)
+        {
+            if (i == currentSelectedRadialPart)
+            {
+                spawnedParts[i].GetComponent<Image>().color = Color.yellow;
+                spawnedParts[i].transform.localScale = 1.1f * Vector3.one;
+                spawnedParts[i].gameObject.transform.Find("TextField").gameObject.MustGetComponent<TextMeshProUGUI>().color = Color.yellow;
+                spawnedParts[i].gameObject.transform.Find("TextField").gameObject.MustGetComponent<TextMeshProUGUI>().fontStyle = (FontStyles)FontStyle.Bold;
+            }
+            else
+            {
+                spawnedParts[i].GetComponent<Image>().color = Color.white;
+                spawnedParts[i].transform.localScale = Vector3.one;
+                spawnedParts[i].gameObject.transform.Find("TextField").gameObject.MustGetComponent<TextMeshProUGUI>().color = Color.black;
+                spawnedParts[i].gameObject.transform.Find("TextField").gameObject.MustGetComponent<TextMeshProUGUI>().fontStyle = (FontStyles)FontStyle.Normal;
+            }
+        }
+    }
+    /// <summary>
+    /// Is true if the radial menu is open or not.
```

Comment:
This makes it sound like the variable should always be true. I assume the below is the intended meaning, right?

```suggestion
    /// Is true if the radial menu is open.
```